services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/document/DefaultEnhancedDocument.java

```diff
@@ -0,0 +1,559 @@
+
+    @Override
+    public <T extends Number> Map<String, T> getMapOfNumbers(String attributeName, Class<T> valueType) {
+        AttributeValue attributeValue = attributeValueMap.get(attributeName);
+        if (attributeValue == null || !attributeValue.hasM()) {
+            return null;
+        }
+        Map<String, T> result = new LinkedHashMap<>();
+        attributeValue.m().entrySet().forEach(
+            entry -> result.put(entry.getKey(),
+                                attributeConverterProviders.converterFor(
+                                    EnhancedType.of(valueType)).transformTo(entry.getValue())));
+        return result;
+    }
+
+    @Override
+    public Map<String, Object> getRawMap(String attributeName) {
+        AttributeValue attributeValue = attributeValueMap.get(attributeName);
+        if (attributeValue == null || !attributeValue.hasM()) {
+            return null;
+        }
+        return toSimpleMapValue(attributeValue.m());
+    }
+
+    @Override
+    public EnhancedDocument getMapAsDocument(String attributeName) {
+        AttributeValue attributeValue = attributeValueMap.get(attributeName);
+        if (attributeValue == null) {
+            return null;
+        }
+        if (!attributeValue.hasM()) {
+            throw new RuntimeException("Cannot get "
+                                       + attributeName
+                                       + " attribute as map since its of type "
+                                       + attributeValue.type());
+        }
+        return new DefaultEnhancedDocument(attributeValue.m());
+    }
+
+    @Override
+    public String getJson(String attributeName) {
+
+        if (attributeValueMap.get(attributeName) == null) {
+            return null;
+        }
+        JsonNode jsonNode = JSON_ITEM_ATTRIBUTE_CONVERTER.transformTo(attributeValueMap.get(attributeName));
+        Document document = jsonNode.visit(new DocumentUnmarshaller());
+        return document.toString();
+    }
+
+    @Override
+    public boolean getBoolean(String attributeName) {
+        return getBool(attributeName);
+    }
+
+    /**
+     * Keeping the backward compatibility with older version of sdk where 0 and 1 are treated a true and false respectively.
+     */
+    private Boolean getBool(String attributeName) {
+        Object object = get(attributeName);
+        if (object instanceof Boolean) {
+            return (Boolean) object;
+        }
+        if (object instanceof String || object instanceof SdkNumber) {
+            if ("1".equals(object.toString())) {
+                return true;
+            }
+            if ("0".equals(object.toString())) {
+                return false;
+            }
+            return Boolean.valueOf((String) object);
+        }
+        throw new IllegalStateException("Value of attribute " + attributeName + " of type " + getTypeOf(attributeName)
+                                        + " cannot be converted into a Boolean value.");
+    }
```

Comment:
BooleanAttributeConverter handles this already. We should be using attribute converters for every conversion we do.